src/geode/georepresentation/core/vertex_identifier.cpp

```diff
@@ -189,6 +199,28 @@ namespace geode
                 } );
         }
 
+        void filter_component_vertices( const uuid& component_id )
+        {
+            for( auto uv_id : Range( nb_unique_vertices() ) )
+            {
+                auto& mesh_component_vertices =
+                    component_vertices_->value( uv_id );
+
+                std::vector< bool > to_keep(
+                    mesh_component_vertices.size(), true );
+                for( auto i : Range( mesh_component_vertices.size() ) )
+                {
+                    if( mesh_component_vertices[i].component_id.id()
+                        == component_id )
+                    {
+                        to_keep[i] = false;
+                    }
+                }
+                mesh_component_vertices =
```

Comment:
Maybe call this only if you went on line 216

Reply:
Good idea. I also add checks in functions extract/delete_vector_elements to escape if nothing to do.